common/djangoapps/student/models.py

```diff
@@ -1793,6 +1793,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user).exists():
+                # if user has not profile then create it.
+                user_profile = UserProfile(name=user.username, user=user)
+                user_profile.save()
+
```

Comment:
Also, these lines should be replaced with:
`user_profile, __ = UserProfile.objects.get_or_create(name=user.username, user=user)`